python/lsst/obs/cfht/rawFormatter.py

```diff
@@ -136,4 +136,8 @@ def readMetadata(self):
 
     def readImage(self):
         index, metadata = self._determineHDU(self.dataId["detector"])
-        return lsst.afw.image.ImageI(self.fileDescriptor.location.path, index)
+        # Post-ISR data from Terrapix Elixir has DATEPROC and EL_SYS keys.
+        if 'DATEPROC' in metadata and 'EL_SYS' in metadata:
```

Comment:
I'm pretty sure that if you use `lsst.afw.image.ImageFitsReader` to read the image you will get the native data type (taking into account scaling) without having to key off these headers.

Reply:
I think something like:

```
reader = lsst.afw.image.ImageFitsReaders(self.fileDescriptor.location.path, hdu=index)
return reader.read()
```